main.go

```diff
@@ -56,6 +57,13 @@ func main() {
 		log.Fatalf("Improperly configured %s", err)
 	}
 
+	valid, errs := config.ValidateConf(conf.YamlConfig, true)
+	if !valid {
+		fmt.Printf("YAML validation - errors: %q", errs)
+		os.Exit(2)
```

Comment:
Maybe instead of magic values use a constant like `const INVALID_CONFIG_ERROR = 2`, and then you can write a more self-describing code, like  `os.Exit(INVALID_CONFIG_ERROR)`

Reply:
Marcin loves returning magic exit codes :) But I'm always against it unless it's documented somewhere. I guess it won't be so just use 1 everywhere.

Reply:
Yeah, exit codes it's old style ;), but when we running software with `supervisor` it's still informative style (In logs we have "crash" code number).

Reply:
It's not old style. And I don't know what's informative in "exit code 2".

Reply:
It shouldn't matter what we are running the software with, you can always do `const BadConfigError = 2` and then `Exit(BadConfigError)`

Reply:
But if we do it that way, then:
- where's documentation for possible error codes?
- are we sure that all possible config errors end up with 2?
- are we sure that not other error case returns 2?
- do we have tests for error codes?

Reply:
It would be helpful during code reviews and reading code in general. If it's defined as a constant then we don't need to guess based on the context what caused the error.

Reply:
I understand, but I feel that this approach does not take error codes seriously. No documentation, no proof they don't repeat, etc. My point of view is that we take it seriously (my questions above) or return 1 everywhere.

Reply:
True dat